### PR TITLE
Update free-courses-en.md

### PR DESCRIPTION
**Pull request by ak191201**
Added Dynamic Programming course

## What does this PR do?
Add resource(s) 

### Description
This is a dynamic programming course by a Google engineer, William Fiset. The course is very articulate and covers dynamic programming for coding interviews.

### Why is this valuable (or not)?
The course is valuable because the instructor is reliable and the content is top notch.

### How do we know it's really free?
This course is available on YouTube as a playlist so it's free.

### For book lists, is it a book? For course lists, is it a course? etc.
It is an online course freely available on YouTube.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
please resolve conflicts.

**Comment by ak191201** [comment]
Conflict resolved

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -141,7 +141,8 @@
 * [Data Structures and Algorithms in Python Full Course for Beginners](https://www.youtube.com/watch?v=pkYVOmU3MgA) - Aakash N S (freeCodeCamp)
 * [Data Structures and Algorithms Specialization](https://www.coursera.org/specializations/data-structures-algorithms) - UC San Diego, HSE University
 * [Data Structures Easy to Advanced Course - Full Tutorial from a Google Engineer](https://www.youtube.com/watch?v=RBSGKlAvoiM) - William Fiset (freeCodeCamp)
-* [Dynamic Programming](https://www.youtube.com/playlist?list=PLMCXHnjXnTnto1pZVvH7rbZ9W5neZ7Yhc) - Gaurav Sen
+* [Dynamic Programming by Gaurav Sen](https://www.youtube.com/playlist?list=PLMCXHnjXnTnto1pZVvH7rbZ9W5neZ7Yhc) - Gaurav Sen
```
Comment (eshellman) [inline courses/free-courses-en.md:144]:
```suggestion
* [Dynamic Programming](https://www.youtube.com/playlist?list=PLMCXHnjXnTnto1pZVvH7rbZ9W5neZ7Yhc) - Gaurav Sen
``` 
Don't invent titles! Use the title given by the resource.

**Review by eshellman — CHANGES_REQUESTED** [review]
adding the author is sufficient to disambiguate the two resources.

**Comment by ak191201** [comment]
Changes made

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -142,6 +142,7 @@
 * [Data Structures and Algorithms Specialization](https://www.coursera.org/specializations/data-structures-algorithms) - UC San Diego, HSE University
 * [Data Structures Easy to Advanced Course - Full Tutorial from a Google Engineer](https://www.youtube.com/watch?v=RBSGKlAvoiM) - William Fiset (freeCodeCamp)
 * [Dynamic Programming](https://www.youtube.com/playlist?list=PLMCXHnjXnTnto1pZVvH7rbZ9W5neZ7Yhc) - Gaurav Sen
+* [Dynamic Programming by William Fiset](https://www.youtube.com/playlist?list=PLDV1Zeh2NRsAsbafOroUBnNV8fhZa7P4u) - WilliamFiset
```
Comment (eshellman) [inline courses/free-courses-en.md:145]:
```suggestion
* [Dynamic Programming](https://www.youtube.com/playlist?list=PLDV1Zeh2NRsAsbafOroUBnNV8fhZa7P4u) - WilliamFiset
```

**Pull request merged**